src/katsdpcontroller/tasks.py

```diff
@@ -736,6 +739,11 @@ async def unhandled_request(self, ctx, req: aiokatcp.core.Message) -> None:
         ctx.reply(aiokatcp.core.Message.OK)
         await ctx.drain()
 
+    def get_command_value(self, value_type: Callable[..., _T], parameter_string: str) -> _T:
```

Comment:
I'd remove "string" from the name - we don't generally include the type of an argument in its name. And I might rename "parameter" to "argument" (since that's what Python calls command-line arguments e.g. in `argparse`). Then while I'm bikeshedding, what do you think about calling the method `get_command_argument`?

Reply:
Sure, `argument_name` sounds good. I was trying to think of a way to name it that makes it obvious that it should include the leading `--`, but I can't think of anything both succinct and clear.

Reply:
BTW if you do keep `value_type`, it can probably be more strictly typed as `Callable[[str], _T]`.